src/cache.ts

```diff
@@ -47,8 +53,30 @@ export const getQuery = async (query: DataSourceRequestOptions, backendSrv: Back
   return promise;
 };
 
+const assetTimeseries = new Map<string, TimeSeriesResponseItem[]>();
+
+export const getTimeseries = (options: QueryOptions, target: QueryTarget) => {
+  const id: string = `${options.dashboardId}_${options.panelId}_${target.refId}_${
```

Comment:
a hashTimeseries object or similiar would be nice, this looks prone to error (and is used multiple places) ;-)

Reply:
added a timeseriesHash() function